src/claim/authorize-ksign-secp256k1/authorize-ksign-secp256k1.js

```diff
@@ -0,0 +1,101 @@
+const snarkjs = require('snarkjs');
+const Entry = require('../entry/entry');
+const utils = require('../../utils');
+const CONSTANTS = require('../../constants');
+
+const { bigInt } = snarkjs;
+const helpers = require('../../sparse-merkle-tree/sparse-merkle-tree-utils');
+
+/**
+ * Class representing an authorized Ksign claim with elliptic curve as secp256k1
+ * Authorized KsignSecp256k1 claim is used to authorize a public key that belongs to elliptic curve secp256k1 for being used afterwards
+ * Authorized KsignSecp256k1 element representation is as follows:
+ * |element 3|: |empty|public key|version|claim type| - |18 bytes|2 bytes|4 bytes|8 bytes|
+ * |element 2|: |empty| - |1 bytes|31 bytes|
+ * |element 1|: |empty| - |32 bytes|
+ * |element 0|: |empty| - |32 bytes|
+ */
+class AuthorizeKSignSecp256k1 {
+  /**
+   * Initialize raw claim data structure
+   * Bytes are taken according element claim structure
+   * Claim type is used to define this concrete claim. This parameter takes 8 bytes.
+   * @param {Object} data - Input parameters
+   * Data input object contains:
+   * {Uint32} _version - Version assigned to the claim
+   * {Bool} _sign - Sign of the coordinate X of an eliptic curve point
+   * {String} _ay - Coordinate Y of an eliptic curve point
+   */
+  constructor(data) {
+    const versionBuff = Buffer.alloc(4);
+    const {
+      version, pubKeyCompressed,
+    } = data;
+    // Save pubKey as compressed
+    versionBuff.writeUInt32BE(version);
+    this._structure = {
+      claimType: helpers.bigIntToBuffer(bigInt(CONSTANTS.CLAIMS.AUTHORIZE_KSIGN_SECP256K1.TYPE)).slice(24, 32),
+      version: versionBuff,
+      pubKeyCompressed: utils.hexToBytes(pubKeyCompressed),
+    };
+  }
+
+  /**
+   * Retrieve claim structure
+   * @returns {Object} Raw data claim structure
+   */
+  get structure() {
+    return this._structure;
+  }
+
+  /**
+   * Code raw data claim object into an entry claim object
+   * @returns {Object} Element representation of the claim
+   */
+  createEntry() {
+    const claimEntry = new Entry();
+    let endIndex = claimEntry.elements[3].length;
+    let startIndex = endIndex - this._structure.claimType.length;
+    // claim element 3 composition
+    claimEntry.elements[3].fill(this._structure.claimType, startIndex, endIndex);
+    endIndex = startIndex;
+    startIndex = endIndex - this._structure.version.length;
+    claimEntry.elements[3].fill(this._structure.version, startIndex, endIndex);
+    let indexLen = this._structure.pubKeyCompressed.length;
```

Comment:
I don't think that you need indexLen variable to recalculate indexLen each time.
you could do it like this to be faster and even more readable:

let indexLen = this._structure.pubKeyCompressed.length;

And then play with it:

this._structure.pubKeyCompressed.slice(indexLen - 2, indexLen)

Since this._structure.pubKeyCompressed is not changing dynamically its lenght, right?

You'll save memory and some performance :)